File: goweb-spring/src/main/java/com/example/goweb_spring/model/ChatMessage.java
package com.example.goweb_spring.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data // Generates getters, setters, equals, hashCode, and toString methods
@NoArgsConstructor // Generates a no-args constructor
@AllArgsConstructor // Generates an all-args constructor
public class ChatMessage {
    private String sender;
    private String content;
    private String roomId;
    private MessageType type;

    public enum MessageType {
        CHAT,
        JOIN,
        LEAVE
    }
}

File: goweb/src/types/ChatMessage.ts
export interface ChatMessage {
  sender: string;
  senderUsername: string;
  content: string;
  roomId: string;
  type: MessageType;
  timestamp: number;
}

export enum MessageType {
  CHAT = "CHAT",
  JOIN = "JOIN",
  LEAVE = "LEAVE",
}
